pkg/admissioncontroller/webhook/auth/seed/authorizer.go

```diff
@@ -256,7 +256,19 @@ func (a *authorizer) authorizeEvent(log logr.Logger, attrs auth.Attributes) (aut
 	return auth.DecisionAllow, "", nil
 }
 
-func (a *authorizer) authorizeLease(log logr.Logger, seedName string, attrs auth.Attributes) (auth.Decision, string, error) {
+func (a *authorizer) authorizeLease(log logr.Logger, seedName string, userType seedidentity.UserType, attrs auth.Attributes) (auth.Decision, string, error) {
+	// extension clients may only work with leases in the seed namespace
+	if userType == seedidentity.UserTypeExtension {
+		if attrs.GetNamespace() == gardenerutils.ComputeGardenNamespace(seedName) {
+			if ok, reason := a.checkVerb(log, attrs, "create", "get", "list", "watch", "update", "patch", "delete", "deletecollection"); !ok {
+				return auth.DecisionNoOpinion, reason, nil
+			}
+
+			return auth.DecisionAllow, "", nil
+		}
+		return auth.DecisionNoOpinion, "lease object is not in seed namespace", nil
+	}
```

Comment:
Should we at least prevent extension clients from tampering with the `gardenlet-leader-election` `Lease`?

Reply:
The `gardenlet-leader-election` `Lease` is located in the `garden` namespace. Hence, the authorizer doesn't allow extensions to tamper with it.

Do you suggest returning `DecisionDeny` explicitly?

Reply:
OK, no then it's fine I guess. Thank you!